CMakeLists.txt

```diff
@@ -6,8 +6,8 @@ find_package(OpenMP REQUIRED)
 
 include_directories(.)
 
-set(CMAKE_CXX_FLAGS_DEBUG "-g ")
-set(CMAKE_CXX_FLAGS_RELEASE "-O3 ")
+set(CMAKE_CXX_FLAGS_DEBUG "-g -Wfatal-errors")
```

Comment:
I would also add a `-Wall`. I thought it was there, but somehow it disappeared during refactorings.